static/src/javascripts/projects/common/modules/discussion/comments.js

```diff
@@ -0,0 +1,588 @@
+
+        window.location.replace(`#comment-${comment.id}`);
+    }
+
+    replyToComment(e: Event): void {
+        e.preventDefault();
+
+        const replyLink: HTMLElement = (e.currentTarget: any);
+        const replyToId = replyLink.getAttribute('data-comment-id') || '';
+        const replyTo = document.getElementById(`reply-to-${replyToId}`);
+
+        // There is already a comment box for this on the page
+        if (replyTo) {
+            replyTo.focus();
+            return;
+        }
+
+        $('.d-comment-box--response').remove();
+
+        const replyToComment = qwery(`#comment-${replyToId}`)[0];
+        const replyToAuthor = replyToComment.getAttribute(
+            'data-comment-author'
+        );
+        const replyToAuthorId = replyToComment.getAttribute(
+            'data-comment-author-id'
+        );
+        const $replyToComment = bonzo(replyToComment);
+        const replyToBody = qwery(
+            this.getClass('commentBody'),
+            replyToComment
+        )[0].innerHTML;
+        const replyToTimestamp = qwery(
+            this.getClass('commentTimestampJs'),
+            replyToComment
+        )[0].innerHTML;
+        const commentBox = new CommentBox({
+            discussionId: this.options.discussionId,
+            premod: this.user.privateFields.isPremoderated,
+            state: 'response',
+            replyTo: {
+                commentId: replyToId,
+                author: replyToAuthor,
+                authorId: replyToAuthorId,
+                body: replyToBody,
+                timestamp: replyToTimestamp,
+            },
+            focus: true,
+        });
+
+        // this is a bit toffee, but we don't have .parents() in bonzo
+        const parentCommentEl = $replyToComment.hasClass(
+            this.getClass('topLevelComment', true)
+        )
+            ? $replyToComment[0]
+            : $replyToComment.parent().parent()[0];
+
+        const showRepliesElem = qwery(
+            this.getClass('showReplies'),
+            parentCommentEl
+        );
+
+        if (
+            showRepliesElem.length > 0 &&
+            !bonzo(showRepliesElem).hasClass('u-h')
+        ) {
+            showRepliesElem[0].click();
+        }
+
+        commentBox.render(parentCommentEl);
+
+        commentBox.on('post:success', (comment: commentType) => {
+            let responses = qwery('.d-thread--responses', parentCommentEl)[0];
+
+            if (!responses) {
+                responses = bonzo.create(
+                    '<ul class="d-thread d-thread--responses"></ul>'
+                )[0];
+                bonzo(parentCommentEl).append(responses);
+            }
+
+            this.destroy();
+            this.addComment(comment, responses);
+        });
+    }
+
+    reportComment(e: Event): void {
+        e.preventDefault();
+
+        const currentTarget: HTMLElement = (e.currentTarget: any);
+        const commentId = currentTarget.getAttribute('data-comment-id') || '';
```

Comment:
If `commentid` is an empty string then presumably this will result in `reportContainer` being null?   

Would it be better to wrap the code below in an `if (commentId) {}` or do `if (!commentId) { return; }` beneath this `commentId` declaration?